## Problem Statement 1:

### Blood glucose levels for obese patients have a mean of 100 with a standard deviation of 15.A researcher thinks that a diet high in raw cornstarch will have a positive effect on blood glucose levels. A sample of 36 patients who have tried the raw cornstarch diet have a mean glucose level of 108. Test the hypothesis that the raw cornstarch had an effect or not.

In [66]:
import math
no_of_sample=36
sample_mean=108
population_mean=100
population_sigma=15

##### Step-1: State the hypotheses. The population mean is 100.
##### H0: μ=100 ==> null hypothesis.
##### H1: μ≠100  ==> Alternate hypothesis.
##### Step-2: Set up the significance level. It is not given in the problem so let’s assume it as 5% (0.05).
##### Step-3: Calculate Z Score.

In [67]:
z=(sample_mean-population_mean)/(population_sigma/math.sqrt(no_of_sample))
print("Z Score =", z)

Z Score = 3.2


##### By looking at z-table and p-value associated with 3.20 is 0.9993.
##### The probability of having value less than 108 is 0.9993 and more than or equals to 108 is (1-0.9993)=0.0007.
##### Step-4: Sice the probability of having mean glucose level more than or equals to 108  is 0.0007 which is less than 0.05. 
##### So we will reject the Null hypothesis i.e. there is raw cornstarch effect.

## Problem Statement 2:

### In one state, 52% of the voters are Republicans, and 48% are Democrats. In a second state, 47% of the voters are Republicans, and 53% are Democrats. Suppose a simple random sample of 100 voters are surveyed from each state.

### What is the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state?

##### P1=the proportion of Republican voters in the first state.
##### P2=the proportion of Republican voters in the second state.
##### p1=the proportion of Republican voters in the sample from the first state.
##### p2=the proportion of Republican voters in the sample from the second state. 

##### The number of voters sampled from the first state (n1)=100.

In [68]:
n1=100

##### The number of voters sampled from the second state (n2)=100.

In [69]:
n2=100

In [70]:
P1=0.52

##### (1-P1)=Q1

In [71]:
Q1=0.48
P2=0.47

##### (1-P2)=Q2

In [72]:
Q2=0.53

##### The mean of the difference in sample proportions i.e. Expected Value E[p1-p2]=P1-P2=mu

In [73]:
mu=P1-P2
std=math.sqrt((P1*(1-P1)/n1)+(P2*(1-P2)/n2))

##### The standard deviation of the difference (std).

In [74]:
print("Mu = "+"{0:.2f}".format(mu))
print("Std = "+"{0:.2f}".format(std))

Mu = 0.05
Std = 0.07


##### This problem requires us to find the probability that p1 is less than p2. 
##### This is equivalent to finding the probability that p1-p2<0.

In [75]:
x=0

##### To find this probability, we need to transform the random variable (p1-p2) into a z-score. 
##### That transformation appears below:

In [76]:
Z_p1_p2 = (x-mu)/std
print("Z_score (p1, p2) = "+"{0:.2f}".format(Z_p1_p2))

Z_score (p1, p2) = -0.71


##### From Z table we find that the probability of a z-score being -0.7082 or less is 0.24.
##### Therefore, the probability that the survey will show a greater percentage of Republican voters in the second state than in the first state is 0.24.

## Problem Statement 3:

### You take the SAT and score 1100. The mean score for the SAT is 1026 and the standard deviation is 209. How well did you score on the test compared to the average test taker?

In [77]:
x=1100
mu=1026
sd=209
z=(x-mu)/sd
print("Z Score = "+"{0:.2f}".format(z))

Z Score = 0.35


##### The above calculation shows that my score is 0.35 standard deviations above the mean.

In [78]:
print("Score is in the range {} - {}  with a  z-score of {:.2f}".format(mu-sd, mu+sd, z))

Score is in the range 817 - 1235  with a  z-score of 0.35
